src/rest/connector/libs/webex/implementation.py

```diff
@@ -0,0 +1,381 @@
+                ret = func(self, *args, **kwargs)
+            return ret
+
+        return decorated
+
+    @BaseConnection.locked
+    def _request(self, method, dn, **kwargs):
+        """ Wrapper to send REST command to device
+
+        Args:
+            method (str): session request method
+
+            dn (str): rest endpoint
+
+        Returns:
+            response.json() or response.text
+
+        Raises:
+            RequestException if response is not ok
+
+        """
+        if not self.connected:
+            raise Exception("'{d}' is not connected for alias '{a}'".format(
+                d=self.device.name, a=self.alias))
+
+        # Deal with the dn
+        full_url = '{f}{dn}'.format(f=self.url, dn=dn)
+
+        if 'data' in kwargs:
+            p = kwargs['data']
+        elif 'json' in kwargs:
+            p = kwargs['json']
+        else:
+            p = ''
+
+        expected_return_code = kwargs.pop('expected_return_code', None)
+
+        log.info("Sending {method} command to '{d}':"
+                 "\nDN: {furl}\nPayload:{p}".format(method=method,
+                                                    d=self.device.name,
+                                                    furl=full_url,
+                                                    p=p))
+
+        # Send to the device
+        response = self.session.request(method=method, url=full_url, **kwargs)
+
+        # An expected return code was provided. Ensure the response has this code.
+        if expected_return_code:
+            if response.status_code != expected_return_code:
+                raise RequestException(
+                    "'{c}' result code has been returned for '{d}'.\n"
+                    "Expected '{expected_c}' result code.\n"
+                    "Response from server: {r}".format(
+                        c=response.status_code,
+                        d=self.device.name,
+                        expected_c=expected_return_code,
+                        r=response.text))
+        else:
+            # No expected return code provided. Make sure it was successful.
+            try:
+                response.raise_for_status()
+            except Exception:
+                raise RequestException("'{c}' result code has been returned "
+                                       "for '{d}'.\nResponse from server: "
+                                       "{r}".format(d=self.device.name,
+                                                    c=response.status_code,
+                                                    r=response.text))
+
+        log.info("Response from '{dev}':\n"
+                 "Result Code: {c}\n"
+                 "Response: {r}".format(dev=self.device.name,
+                                        c=response.status_code,
+                                        r=response.text))
+
+        # In case the response cannot be decoded into json
+        # warn and return the raw text
+        if response.text:
```

Comment:
I found in case of `DELETE`, it will return `204` with empty string. in that case, it warns `Could not decode json. Return text`. So, I detoured the case to return as is without warning since it's expected.
```
[2021-03-13, 10:24:09.216 AM] Sending DELETE command to 'webex':
[2021-03-13, 10:24:09.216 AM] DN: https://api.ciscospark.com:443/v1/messages/Y2lzY29zcGFy************TNjODMzYTM2
[2021-03-13, 10:24:09.216 AM] Payload:
[2021-03-13, 10:24:09.463 AM] Response from 'webex':
[2021-03-13, 10:24:09.463 AM] Result Code: 204
[2021-03-13, 10:24:09.463 AM] Response:
[2021-03-13, 10:24:09.463 AM] Could not decode json. Returning text!
[2021-03-13, 10:24:09.468 AM] Out[12]: ''